demos/nordic/nrf52840-dk/bootloader/main.c

```diff
@@ -240,21 +246,31 @@ void vCommitSecondBank()
                           BANK2_DESCRIPTOR->ulStartAddress );
 }
 
+
+#ifdef  __GNUC__ 
+#pragma GCC push_options
+#pragma GCC optimize ("O0")
+#endif
+int vBlinkLeds (){
+    nrf_gpio_cfg_output(LED_1);
+    nrf_gpio_cfg_output(LED_2);
+    nrf_gpio_pin_clear(LED_2);
+    nrf_delay_ms(100);
+    for (int i = 0; i < 6; i++) {
+        nrf_gpio_pin_toggle(LED_1);
+        nrf_delay_ms(100);
+    }
+    return 0;
+}
+#ifdef  __GNUC__ 
+#pragma GCC pop_options
+#endif
+
 /**@brief Function for application main entry. */
 int main( void )
 {
     volatile uint32_t xErrCode = 0;
-
-    /* Protect MBR and bootloader code from being overwritten. */
-    xErrCode = nrf_bootloader_flash_protect( 0, MBR_SIZE, false );
-    APP_ERROR_CHECK( xErrCode );
-    xErrCode = nrf_bootloader_flash_protect( BOOTLOADER_START_ADDR,
-                                             BOOTLOADER_SIZE, false );
-    APP_ERROR_CHECK( xErrCode );
-
-    /*    uint32_t err_code = sd_softdevice_disable(); */
-    /*    APP_ERROR_CHECK(err_code); */
-
+    vBlinkLeds();
     bool bBank1AFRHeader =
         ( bValidateHeader( ( uint8_t * ) CODE_REGION_1_START ) == NRF_SUCCESS );
 
```

Comment:
xErrCode = bValidateHeader( ( uint8_t * ) CODE_REGION_2_START );
=> Maybe create a bBank2AFRHeader  for consistency sake

Reply:
Done